src/vs/workbench/contrib/terminal/test/browser/xterm/shellIntegrationAddon.test.ts

```diff
@@ -211,6 +211,57 @@ suite('ShellIntegrationAddon', () => {
 			mock.verify();
 		});
 	});
+	suite('BufferMarkCapability', async () => {
+		test('SetMark', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;\x07');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), true);
+		});
+		test('SetMark - ID', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;1;\x07');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), true);
+		});
+		test('SetMark - hidden', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;;Hidden\x07');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), true);
+		});
+		test('SetMark - hidden & ID', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;1;Hidden\x07');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), true);
+		});
+		test('SetMark - invalid', async () => {
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, 'foo');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
+			await writeP(xterm, '\x1b]633;SetMark;;;\x07');
+			strictEqual(capabilities.has(TerminalCapability.BufferMarkDetection), false);
```

Comment:
It's currently very strict about the param count but not strict at all about the validity of the params, we should go one of 2 routes:

- Fail on anything unexpected
- Ignore unrecognized params

I think the latter is probably best which means we should remove the <= 2 param assertion.

Reply:
ok will defer and write tests for those cases